assembly/src/library/tests.rs

```diff
@@ -154,20 +154,13 @@ fn library_procedure_collision() -> Result<(), Report> {
         .with_library(lib1)?
         .assemble_library([bar])?;
 
-    let lib2_bar_bar1 = QualifiedProcedureName::from_str("lib2::bar::bar1").unwrap();
-    let lib2_bar_bar2 = QualifiedProcedureName::from_str("lib2::bar::bar2").unwrap();
-
     // make sure lib2 has the expected exports (i.e., bar1 and bar2)
     assert_eq!(lib2.num_exports(), 2);
 
-    // make sure that bar1 and bar2 are equal nodes in the MAST forest
-    assert_eq!(lib2.get_export_node_id(&lib2_bar_bar1), lib2.get_export_node_id(&lib2_bar_bar2));
```

Comment:
Ah - I see! There is a way to solve it - though, not if it makes things more complicated or not. We can define `eq_hash` as a struct:
```rust
pub struct EqHash {
    mast_root: RpoDigest,
    decorator_root: Option<Blake3Digest>,
}
```
`decorator_root` would be computed very similarly to how we compute `eq_hash` now except that if all child nodes have it set to `None`, then the parent node would also have it set to `None`.

This way, both `External` and `Join` node will end up with the same `EqHash` and this would work as before.

Reply:
Good idea! Implemented.